apps/demo-ui/ui/src/Chatbox.tsx

```diff
@@ -927,6 +928,20 @@ export const ChatBox = ({ chatHistory, searchResults, setChatHistory, setSearchR
                 setDocsLoading(false)
             }
             const startTime = new Date(Date.now());
+            const truncateString = (chatInput: string, maxLength = 15) => {
+                if (chatInput.length > maxLength) {
+                  return chatInput.slice(0, maxLength) + '...';
+                } else {
+                  return chatInput;
+                }
+              }
+            if(!settings.activeConversation) {
+                const conversationId = await createConversation(truncateString(chatInput));
+                settings.activeConversation = conversationId.memory_id;
+                setSettings(settings);
```

Comment:
tbh this is more of a "Henry doesn't understand react" comment than anything else - 
what does `setSettings(settings)` do? Is the `settings` object not referred to by a whole bunch of places, so just doing the `settings.activeConversations = blah` updates their state as well? I guess you need to tell the other referrers that their state has been updated and that's what `setSettings` does?

Reply:
`settings` is one of the states of our app declared with the help of useState react hook.
`const [settings, setSettings] = useState(new Settings());`
setSettings is a state updater function provided by the useState hook.
The primary purpose of setSettings in this scenario isn't just about updating the activeConversation. It's about telling React that a state change has occurred, which is necessary to trigger the re-rendering process.

Technically, I don't need to call setsettings if I just want to update a property within the object. But without setSettings, React wouldn't know to re-render the component, and the UI wouldn't update to reflect the change in activeConversation.